client/data/site-roles/with-site-roles.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import useSiteRolesQuery from './use-site-roles-query';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+const withSiteRoles = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const siteId = useSelector( getSelectedSiteId );
+		const { data } = useSiteRolesQuery( siteId );
```

Comment:
This will send funny REST requests (`/sites/null/roles`) if the `siteId` is null. The existing code prevents these request by rendering the query component only when `siteId` is not null.

With React Query we can use the `disabled` option:
```js
useQuery( 'site-roles', fetchSiteRoles, {
  enabled: siteId != null
} );
```

Reply:
Fixed with https://github.com/Automattic/wp-calypso/pull/53503/commits/5718d31ee4f40b05389777aafba09bb268f597e5